devicetypes/zooz/zooz-remote-switch-zen34.src/zooz-remote-switch-zen34.groovy

```diff
@@ -0,0 +1,321 @@
+	logDebug "configure()..."
+	List<String> cmds = []
+
+	if (state.refreshAll || !device.currentValue("firmwareVersion")) {
+		cmds << secureCmd(zwave.versionV1.versionGet())
+	}
+
+	if (state.refreshAll || !device.currentValue("battery")) {
+		cmds << secureCmd(zwave.batteryV1.batteryGet())
+	}
+
+	state.refreshAll = false
+
+	if (state.wakeUpInterval != wakeUpInterval) {
+		cmds << secureCmd(zwave.wakeUpV2.wakeUpIntervalSet(seconds: wakeUpInterval, nodeid: zwaveHubNodeId))
+		cmds << secureCmd(zwave.wakeUpV2.wakeUpIntervalGet())
+	}
+
+	configParams.each { name, param ->
+		Integer storedVal = getStoredVal(name)
+		Integer settingVal = getSettingVal(name)
+		if (storedVal != settingVal) {
+			logDebug "Changing ${param.title}(#${param.num}) from ${storedVal} to ${settingVal}"
+			cmds << secureCmd(zwave.configurationV1.configurationSet(parameterNumber: param.num, size: param.size, scaledConfigurationValue: settingVal))
+			cmds << secureCmd(zwave.configurationV1.configurationGet(parameterNumber: param.num))
+		}
+	}
+	if (cmds) {
+		sendHubCommand(cmds, 500)
+	}
+}
+
+def ping() {
+	logDebug "ping()..."
+}
+
+def refresh() {
+	logDebug "refresh()..."
+	state.refreshAll = true
+
+	refreshSyncStatus()
+
+	logForceWakeupMessage("The next time the device wakes up, the sensor data will be requested.")
+}
+
+String secureCmd(cmd) {
+	if (zwaveInfo?.zw?.contains("s")) {
+		return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		return cmd.format()
+	}
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description, commandClassVersions)
+	if (cmd) {
+		zwaveEvent(cmd)
+	} else {
+		log.warn "Unable to parse: $description"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpIntervalReport cmd) {
+	logDebug "$cmd"
+	runIn(4, refreshSyncStatus)
+	state.wakeUpInterval = cmd.seconds
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	logDebug "Device Woke Up..."
+	runIn(4, refreshSyncStatus)
+	configure()
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	runIn(4, refreshSyncStatus)
+	String name = configParams.find { name, param -> param.num == cmd.parameterNumber }?.key
+	if (name) {
+		int val = cmd.scaledConfigurationValue
+		state[name] = val
+		logDebug "${configParams[name]?.title}(#${configParams[name]?.num}) = ${val}"
+	} else {
+		logDebug "Parameter #${cmd.parameterNumber} = ${cmd.scaledConfigurationValue}"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
+	logDebug "${cmd}"
+	sendEvent(name: "firmwareVersion", value: (cmd.applicationVersion + (cmd.applicationSubVersion / 100)))
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def val = (cmd.batteryLevel == 0xFF ? 1 : cmd.batteryLevel)
+	if (val > 100) {
+		val = 100
+	}
+	logDebug "Battery is ${val}%"
+	sendEvent(name:"battery", value:val, unit:"%", isStateChange: true)
```

Comment:
No

Reply:
Just to be clear, I have no problem making changes for consistency, readability, optimization, or are necessary to meet ST standards and every time a reviewer requests a change I make sure that I include that change in every DTH I submit going forward.

Using createEvent instead of sendEvent will require me to change the return type of those methods, make the code harder to follow, make it inconsistent with the central scene method, and I've submitted 5 custom DTHs _(3 different manufacturers)_ that only use sendEvent and you approved the other 4 without requesting this change so those are the reasons I said no...